application/models/user.go

```diff
@@ -516,6 +538,18 @@ func (u *User) SetLocation(location Location) error {
 	return u.Save()
 }
 
+// RemoveLocation removes the location record associated with the user
+func (u *User) RemoveLocation() error {
+	if u.LocationID.Valid {
```

Comment:
Unnecessarily nested `if`, but the function is so short that I don't mind if you leave it this way. (-;